dhis-2/dhis-support/dhis-support-db-migration/src/main/resources/org/hisp/dhis/db/migration/2.34/V2_34_10__Migrate_Data_Into_Visualization_Tables.sql

```diff
@@ -0,0 +1,690 @@
+-- This script responsible for migrating required data to Visualization.
+-- See Feature DHIS2-7946
+
+-- It migrates:
+-- 1) copy data from chart table into visualization;
+-- 2) copy data from reporttable table into visualization;
+-- 3) copy reporttable column into the visualizationid column in the dashboarditems table;
+-- 4) copy chartid column into the visualizationid column in the dashboarditems table;
+-- 5) copy the reporttableid column into the visualizationid column in the report table;
+-- 6) copy data from series table into axis table;
+-- 7) copy data from chart_seriesitems table into visualization_axis table;
+-- 8) set default values for NULL (Java primitive) columns.
+
+
+-- Set any possible empty or null NAME to an empty space ' ', as name is mandatory.
+UPDATE chart SET name = ' ' WHERE COALESCE(name, '') = '' OR name IS NULL;
```

Comment:
Is this something we want to support? An empty chart name?

Reply:
Currently, the "name" column is not mandatory.
But now, I am making it mandatory in the new Visualization table.
So, I am afraid that for some reason we can face some null values, in the existing tables, during the migration which will cause it to fail.
That's the reason for this statement, it's a kind of preventive query...

Reply:
Got it